cmd/fleet/cron.go

```diff
@@ -689,3 +616,119 @@ func newFailerClient(forcedFailures string) *worker.TestAutomationFailer {
 	}
 	return failerClient
 }
+
+func startCleanupsAndAggregationSchedule(
+	ctx context.Context, instanceID string, ds fleet.Datastore, logger kitlog.Logger, enrollHostLimiter fleet.EnrollHostLimiter,
+) {
+	schedule.New(
+		ctx, "cleanups_then_aggregation", instanceID, 1*time.Hour, ds,
+		// Using leader for the lock to be backwards compatilibity with old deployments.
+		schedule.WithAltLockID("leader"),
+		schedule.WithLogger(kitlog.With(logger, "cron", "cleanups_then_aggregation")),
+		// Run cleanup jobs first.
+		schedule.WithJob(
+			"distributed_query_campaings",
+			func(ctx context.Context) error {
+				_, err := ds.CleanupDistributedQueryCampaigns(ctx, time.Now())
+				return err
+			},
+		),
+		schedule.WithJob(
+			"incoming_hosts",
+			func(ctx context.Context) error {
+				_, err := ds.CleanupIncomingHosts(ctx, time.Now())
+				return err
+			},
+		),
+		schedule.WithJob(
+			"carves",
+			func(ctx context.Context) error {
+				_, err := ds.CleanupCarves(ctx, time.Now())
+				return err
+			},
+		),
+		schedule.WithJob(
+			"expired_hosts",
+			func(ctx context.Context) error {
+				_, err := ds.CleanupExpiredHosts(ctx)
+				return err
+			},
+		),
+		schedule.WithJob(
+			"policy_membership",
+			func(ctx context.Context) error {
+				return ds.CleanupPolicyMembership(ctx, time.Now())
+			},
+		),
+		schedule.WithJob(
+			"sync_enrolled_host_ids",
+			func(ctx context.Context) error {
+				return enrollHostLimiter.SyncEnrolledHostIDs(ctx)
+			},
+		),
+		// Run aggregation jobs after cleanups.
+		schedule.WithJob(
+			"query_aggregated_stats",
+			func(ctx context.Context) error {
+				return ds.UpdateQueryAggregatedStats(ctx)
+			},
+		),
+		schedule.WithJob(
+			"scheduled_query_aggregated_stats",
+			func(ctx context.Context) error {
+				return ds.UpdateScheduledQueryAggregatedStats(ctx)
+			},
+		),
+		schedule.WithJob(
+			"aggregated_munki_and_mdm",
+			func(ctx context.Context) error {
+				return ds.GenerateAggregatedMunkiAndMDM(ctx)
+			},
+		),
+		schedule.WithJob(
+			"update_os_versions",
+			func(ctx context.Context) error {
+				return ds.UpdateOSVersions(ctx)
+			},
+		),
```

Comment:
This is a huge nit and might be just a me thing, but I wonder if it'd be more readable like this:

```
		// ...
		schedule.WithJob("sync_enrolled_host_ids", func(ctx context.Context) error { return enrollHostLimiter.SyncEnrolledHostIDs(ctx) }),
		// Run aggregation jobs after cleanups.
		schedule.WithJob("query_aggregated_stats", func(ctx context.Context) error { return ds.UpdateQueryAggregatedStats(ctx) }),
		schedule.WithJob("scheduled_query_aggregated_stats", func(ctx context.Context) error { return ds.UpdateScheduledQueryAggregatedStats(ctx) }),
		schedule.WithJob("aggregated_munki_and_mdm", func(ctx context.Context) error { return ds.GenerateAggregatedMunkiAndMDM(ctx) }),
		schedule.WithJob("update_os_versions", func(ctx context.Context) error { return ds.UpdateOSVersions(ctx) }),
```

I might alone in this preference though :D but couldn't help mentioning it.